Expensify/App
# chore: Improved error messages for sign-in flow

### Details
Currently, when a user encounters a sign-in error on [expensify.cash](https://expensify.cash/), we display the raw API message to the user. In this PR, I have made a utility function to extract the `jsonCode` from the API error response and display the respective error message based on the `jsonCode` defined in the newly added utility function `getErrorMessageFromErrorCode` defined in a new file `src/libs/ErrorUtils.js`.

### Fixed Issues
<Please replace GH_LINK with the link to the GitHub issue this Pull Request is fixing>
Fixes #1091

### Tests
1. Open the Expensify application.
2. `Signup for Waitlist` screen will appear
3. Enter your Email/Phone into the text field
4. Tap on the `Continue Button`
5. `Enter Password` screen will appear
6. Try entering an invalid password
7. A red text with the message `Incorrect login or password. Please try again.` will appear
8. Repeat the same step for all of different use cases defined in the issue #1091 statement

### Tested On


### Screenshots

#### Web
<img width="705" alt="Web" src="https://user-images.githubusercontent.com/76831255/104084135-e2cc8f80-526a-11eb-9c56-cfaaea1edcd3.png">


#### Mobile Web
<img width="415" alt="Screenshot 2021-01-10 at 9 29 29 AM" src="https://user-images.githubusercontent.com/76831255/104114126-db6abc00-5326-11eb-97c5-365c92a1f3f7.png">


#### Desktop
<img width="1089" alt="Desktop" src="https://user-images.githubusercontent.com/76831255/104084138-ec55f780-526a-11eb-9718-9ee110c8e493.png">


#### iOS
![Simulator Screen Shot - iPhone 11 Pro - 2021-01-09 at 11 04 03](https://user-images.githubusercontent.com/76831255/104084143-f4ae3280-526a-11eb-9ccc-48015b1e1907.png)


#### Android
![Screenshot_1610170603](https://user-images.githubusercontent.com/76831255/104084148-fe379a80-526a-11eb-8c66-1736ea3c5579.png)
